arraycontext/impl/pytato/__init__.py

```diff
@@ -47,14 +47,39 @@
 from arraycontext.metadata import NameHint
 
 import numpy as np
-from typing import Any, Callable, Union, TYPE_CHECKING, Tuple, Type
-from pytools.tag import ToTagSetConvertible
+from typing import Any, Callable, Union, TYPE_CHECKING, Tuple, Type, FrozenSet
+from pytools.tag import ToTagSetConvertible, normalize_tags, Tag
 import abc
 
 if TYPE_CHECKING:
     import pytato
 
 
+def _preprocess_array_tags(tags: ToTagSetConvertible) -> FrozenSet[Tag]:
+    tags = normalize_tags(tags)
+
+    name_hints = [tag for tag in tags if isinstance(tag, NameHint)]
```

Comment:
Add vim fold-markers for preprocessing the `NameHint` tags? I believe there would be more such mappings in the near future.

Reply:
https://github.com/inducer/arraycontext/compare/98a3f9116bdc80eccf623edd42ff35ed95b024fa..327103662e4c0c220ae35d12c531f0a5f1254541